packages/backend/src/controllers/messages/message.class.ts

```diff
@@ -1,15 +1,44 @@
-import { FMDBMessageTemplate } from './message.interface';
+import { compile } from 'handlebars';
 
-export class MessageTemplate implements FMDBMessageTemplate {
+/**
+ * Abstract class for notification messages
+ *
+ * @export
+ * @abstract
+ * @class Message
+ */
+export abstract class Message {
 
-  readonly template: string;
+  /**
+   * handlebar.js template for Markdown messages
+   *
+   * @protected
+   * @abstract
+   * @type {Template}
+   * @memberof Message
+   */
+  protected abstract readonly markdownTemplate: string;
 
-  apply(options: any): string {
-    const keys = Object.keys(options);
-    let message = this.template;
-    keys.forEach(key => {
-      message = message.replace('{' + key + '}', options[key]);
-    });
-    return message;
+  /**
+   * Content for the templates
+   *
+   * ! Content depends on the used template class
+   *
+   * @protected
+   * @abstract
+   * @type {*}
+   * @memberof Message
+   */
+  protected context: any;
+
+  /**
+   * Return the message as Markdown
+   *
+   * @readonly
+   * @type {string}
+   * @memberof Message
+   */
+  public get Markdown() : string {
```

Comment:
```suggestion
  public get markdown() : string {
```

Wouldn't the naming convention suggest a getter to start with a lower-case letter?

Reply:
Hm, I do not know 🤔 The naming conventions do not really say something about this, AirBnB (style guide) instead recommend to avoid getter/setters (Ähm ... nope 😆). So I thought it would be a great idea to make a small visual separation from these computed getters compared with other basic properties. In future, there will be getters for:

| PascalCase            | camelCase             |
|-----------------------|-----------------------|
| `message.Embeds`      | `message.embeds`      |
| `message.AdaptiveKit` | `message.adaptiveKit` |
| `message.BlockKit`    | `message.blockKit`    |
| `message.HTML`        | `message.html`        |

What do you @p-fruck think is better? I am not really sure 😅

Reply:
@tjarbo In my opinion getters and setters should be represented as if they were regular attributes, regardless whether they require additional computation or not, which is why I would propose them to be written in camelCase. But this is just my personal opinion, I could be completely wrong about that :laughing:

Reply:
Okay, I agree with you. Renamed it in 7cf5ed5